clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -146,8 +198,9 @@ public void authenticate() throws IOException {
         if (netOutBuffer != null && !flushNetOutBufferAndUpdateInterestOps())
             return;
 
-        if (saslServer.isComplete()) {
+        if (saslServer != null && saslServer.isComplete()) {
             transportLayer.removeInterestOps(SelectionKey.OP_WRITE);
```

Comment:
Do we actually need this `removeInterestOps`? It seems to me that we remove it in `flushNetOutBufferAndUpdateInterestOps` before we reach here.